src/lcm/simulate.py

```diff
@@ -0,0 +1,457 @@
+import inspect
+from functools import partial
+
+import jax.numpy as jnp
+from dags import concatenate_functions
+
+from lcm.argmax import argmax, segment_argmax
+from lcm.dispatchers import spacemap, vmap_1d
+from lcm.interfaces import Space
+
+# ======================================================================================
+# Simulate
+# ======================================================================================
+
+
+def simulate(
```

Comment:
We should write a test for simulate that does not use `get_lcm_function` so we can see if the problem is in the input generation or simulate itself.

Reply:
In `test_simulate.py` I have created the tests:
- `test_simulate()`
- `test_simulate_debug()`
The second one calls `simulate` using `get_lcm_function` and the first one creates the inputs. The tests should test the same behavior (use the same model and parameters); however, only the first one passes. I will look into the details tomorrow.